test/test_opam_cmd.ml

```diff
@@ -40,3 +46,55 @@ let test_tag_from_archive =
     make_test ~archive:"https://other.domain.com/some/path/file-v1.2.3.tbz"
       ~expected:(Some "v1.2.3") ()
   ]
+
+let test_classify_package =
+  let make_test ~name ~package ?(dev_repo = "dummy-dev-repo") ?archive ?(pins = []) ~expected () =
+    let test_name = Printf.sprintf "classify_package: %s" name in
+    let test_fun () =
+      let actual = Duniverse_lib.Opam_cmd.classify_package ~package ~dev_repo ~archive ~pins () in
+      Alcotest.(check (pair Testable.opam_repo (option string))) test_name expected actual
+    in
+    (test_name, `Quick, test_fun)
+  in
+  [ make_test ~name:"base package"
+      ~package:{ name = "ocaml"; version = None }
+      ~expected:(`Virtual, None)
+      ();
+    make_test ~name:"base package versioned"
+      ~package:{ name = "ocaml"; version = Some "1" }
+      ~expected:(`Virtual, None)
+      ();
+    make_test ~name:"empty dev-repo" ~package:{ name = "x"; version = None } ~dev_repo:""
+      ~expected:(`Virtual, None)
+      ();
+    make_test ~name:"no archive" ~package:{ name = "x"; version = None }
+      ~dev_repo:"host.com/path.git" ?archive:None
+      ~expected:(`Virtual, None)
+      ();
+    make_test ~name:"github dev-repo" ~package:{ name = "x"; version = None }
+      ~dev_repo:"git+https://github.com/user/repo.git" ~archive:""
+      ~expected:(`Github ("user", "repo"), None)
+      ();
+    make_test ~name:"guess tag from archive" ~package:{ name = "x"; version = None }
+      ~dev_repo:"git+https://github.com/user/repo.git" ~archive:"file-v1.tbz"
+      ~expected:(`Github ("user", "repo"), Some "v1")
+      ();
+    make_test ~name:"unexpected github url" ~package:{ name = "x"; version = None }
+      ~dev_repo:"git+https://github.com/user/repo/oh/my/what/is/this" ~archive:""
+      ~expected:(`Error "weird github url", None)
+      ();
+    make_test ~name:"no host" ~package:{ name = "x"; version = None } ~dev_repo:"nohost.git"
+      ~archive:""
+      ~expected:(`Error "dev-repo without host", None)
+      ();
+    make_test ~name:"git" ~package:{ name = "x"; version = None }
+      ~dev_repo:"git+https://host.com/some-repo.git" ~archive:"gitpaf#pouf"
+      ~expected:(`Git "https://host.com/some-repo.git", None)
+      ();
+    make_test ~name:"wrong vcs" ~package:{ name = "x"; version = None }
+      ~dev_repo:"hg+https://host.com/some-repo" ~archive:""
+      ~expected:(`Error "dev-repo doesn't use git as a VCS", None)
+      ()
+  ]
```

Comment:
Can you add tests with no "git+" prefix with and without the ".git" extension?

Reply:
These cases are tested in `Test_opam.Dev_repo` already, no point in duplicating them here.